src/plugins/data/public/search/search_interceptor.ts

```diff
@@ -121,6 +129,7 @@ export class SearchInterceptor {
 
       return search(request as any, { ...options, signal: combinedSignal }).pipe(
         takeUntil(timeoutError$),
+        takeUntil(userAbort$),
```

Comment:
Is there a reason we're now handling user abort events separately? If I remember correctly, aborting a search should end up causing an error in the search strategy itself. I guess it might make sense to always throw an error if the search is aborted, but just wanted to make sure this was completely intentional.

Reply:
I handled them separately because I wanted to throw different errors for each one.
I'm open to alternative implementations if they are more simple!